Добавлены Gradle, Lombok, Jackson и логирование. Реализован парсинг J…

Colgens opened the pull request:
…SON и конвертация в XML.

- Подключен Gradle для управления сборкой проекта.
- Добавлена зависимость Lombok для автоматической генерации геттеров, сеттеров и конструктора.
- Подключена зависимость Jackson и создан класс JsonToObject с методом parseJsonToObject для парсинга JSON файлов в объекты.
- Реализован класс Location на основе JSON файла для хранения данных.
- Создан класс ObjectToXmlConverter и его метод toXML() для конвертации объекта в XML формат.
- Добавлено логирование с использованием разных уровней (info, error, debug, warn).

Colgens committed:
Добавлены Gradle, Lombok, Jackson и логирование. Реализован парсинг JSON и конвертация в XML.

- Подключен Gradle для управления сборкой проекта.
- Добавлена зависимость Lombok для автоматической генерации геттеров, сеттеров и конструктора.
- Подключена зависимость Jackson и создан класс JsonToObject с методом parseJsonToObject для парсинга JSON файлов в объекты.
- Реализован класс Location на основе JSON файла для хранения данных.
- Создан класс ObjectToXmlConverter и его метод toXML() для конвертации объекта в XML формат.
- Добавлено логирование с использованием разных уровней (info, error, debug, warn).

kraskden commented on src/main/java/com/tbank/fintech_project/model/entities/Location.java:
```diff
@@ -0,0 +1,13 @@
+
+@Data
+public class Location {
+    @JsonProperty
```
Обычно @JsonProperty ставится если нужно переопределить имя поля в json, без доп параметров эта аннотация излишня

Colgens replied:
> Обычно @JsonProperty ставится если нужно переопределить имя поля в json, без доп параметров эта аннотация излишня

Исправил, ранее без аннотации 'JsonProperty' Jackson не видел полей при десериализации, если сеттер и геттер сгенерированы через Lombok. После добавления зависимости 'annotationProcessor ("org.projectlombok:lombok:1.18.34")' всё заработало как нужно.

Colgens committed:
refactor: Улучшение интеграции Jackson и Lombok

- Добавлена зависимость для обеспечения корректной десериализации объектов Jackson с использованием Lombok.
- Удалены излишние аннотации JsonProperty из классов Location и Coords для упрощения кода.

Colgens committed:
Добавление JUnit для тестов, конструкторов в классах Coords и Location, перенос кода из Main в тесты, добавление Logback для тестового пакета

- Добавлены зависимости JUnit для тестов.
- Добавлены конструкторы с аргументами и без для классов Coords и Location.
- Код из Main разделён и перенесён в тесты.
- Добавлен Logback для тестового пакета.
